src/upgrade/nc_upgrade_scripts/1.0.0/config_dir_restructure.js

```diff
@@ -0,0 +1,206 @@
+/* Copyright (C) 2024 NooBaa */
+'use strict';
+
+const path = require('path');
+const config = require('../../../../config');
+const { ConfigFS } = require('../../../sdk/config_fs');
+const native_fs_utils = require('../../../util/native_fs_utils');
+const nb_native = require('../../../util/nb_native');
+
+/**
+* @typedef {{ 
+    * account_name: String, 
+    * _id: String, 
+    * access_keys: Object[], 
+    * account_old_path: String, 
+    * identity_path: String, 
+    * identity_dir_path: String, 
+    * src_file: nb.NativeFile, 
+    * account_old_path_stat: nb.NativeFSStats, 
+    * gpfs_options?: { src_file: nb.NativeFile, dst_file: nb.NativeFile } 
+* }} AccountUpgradeParams
+*/
+
+/**
+ * run does the following config directory restructure - 
+ * 1. creation of the identities/ directory
+ * 2. creation of accounts_by_name/ directory
+ * 3. Upgrade config files of all accounts under accounts/ (old directory) 
+ * 4. delete accounts/ directory
+ * @param {*} dbg 
+ */
+async function run({ dbg }) {
+    try {
+        const failed_accounts = [];
```

Comment:
you never assign to `failed_accounts`. you return a different one in `upgrade_accounts_config_files`.